document-storage-adapter/src/Pssg.DocumentStorageAdapter/Controllers/FileController.cs

```diff
@@ -86,11 +86,38 @@ public async Task<ActionResult> DownloadInternal( ViewModels.Download download,
             {
                 var _S3 = new S3(_configuration);
                 Dictionary<string, string> metaData = new Dictionary<string, string>();
-                var fileContents = _S3.DownloadFile(fileName, ref metaData);
+
+                byte[] fileContents;
+
+                // if the document swap flag is set, and the filename is .tif, look for a converted PDF
+                if (!String.IsNullOrEmpty(_configuration["SWAP_PDF"]) && fileName.ToLower().Contains(".tif"))
```

Comment:
nit pick: EndsWith will eliminate weird edge cases like "thefile.tiffany.pdf"
`.EndsWith(".tif", StringComparison.CurrentCultureIgnoreCase)`

Reply:
alas sometimes TIF is TIFF